lib/tom_queue/deferred_work_manager.rb

```diff
@@ -113,6 +121,24 @@ def stop
       deferred_set && deferred_set.interrupt
     end
 
-  end
+    # Register handlers to shut down the manager gracefully
+    def register_signal_handlers
+      Thread.main[:signals] = []
+
+      [:QUIT, :INT, :TERM].each do |sig|
+        trap(sig) do
+          Thread.main[:signals] << sig
+        end
+      end
```

Comment:
You shouldn't need to open the mutex to signal the condvar to interrupt any blocked processes - as long as you set the flag _before_ you signal, I _think_ it should be ok :)